spring-ai-openai/src/main/java/org/springframework/ai/openai/client/OpenAiClient.java

```diff
@@ -89,36 +92,46 @@ public AiResponse generate(Prompt prompt) {
 			.temperature(this.temperature)
 			.messages(theoMessages)
 			.build();
+
 		return getAiResponse(chatCompletionRequest);
 	}
 
+	private AiResponse getAiResponse(ChatCompletionRequest chatCompletionRequest) {
+		logger.trace("ChatMessages: {}", chatCompletionRequest.getMessages());
+		ChatCompletionResult chatCompletionResult = this.openAiService.createChatCompletion(chatCompletionRequest);
+		List<ChatCompletionChoice> chatCompletionChoices = chatCompletionResult.getChoices();
+		logger.trace("ChatCompletionChoices: {}", chatCompletionChoices);
+		List<Generation> generations = new ArrayList<>();
+		for (ChatCompletionChoice chatCompletionChoice : chatCompletionChoices) {
+			ChatMessage chatMessage = chatCompletionChoice.getMessage();
+			Generation generation = new Generation(chatMessage.getContent(), Map.of("role", chatMessage.getRole()));
+			generations.add(generation);
+		}
+		return new AiResponse(generations, OpenAiMetadata.from(chatCompletionResult));
```

Comment:
One area for the future is to have some sort of stats collection that goes on that can be sent to a dashboard.  Adding in micrometer and a grafana dashboard could be a relatively easy win to help folks get a handle on costs.